lib/natalie/compiler/pass1.rb

```diff
@@ -2278,6 +2289,20 @@ def transform_rescue_reference_node(node)
           ]
         when ::Prism::GlobalVariableTargetNode
           instructions << GlobalVariableSetInstruction.new(node.name)
+        when ::Prism::IndexTargetNode
+          instructions = [transform_expression(node.receiver, used: true)]
+          # Get rid of the PushArgcInstruction from transform_arguments_node_for_callish
```

Comment:
I've updated this comment a bit, since this is not how I meant it.

The result of `transform_arguments_node_for_callish` are the full instructions when making a call. So this pushes every argument on the stack, and adds a `PushArgcInstruction` with this number of arguments. But we need to include `$!` on the stack too, so that's why we remove the last instruction (the `PushArgc`), push `$!`, and add a new `PushArgc` with the number of arguments plus one.

So getting rid of the `PushArgcInstruction` would be a neat thing, but is not related to this comment.